Fragment.Fragment/MyFlexibleFragment/kotlin/app/src/main/java/com/dicoding/picodiploma/myflexiblefragment/CategoryFragment.kt

```diff
@@ -41,12 +41,12 @@ class CategoryFragment : Fragment(), View.OnClickListener {
             jika ada fragment di dalam backstack maka fragment yang akan di close / remove
             jika sudah tidak ada fragment di dalam backstack maka activity yang akan di close / finish
              */
-            val mFragmentManager = fragmentManager as FragmentManager
-            mFragmentManager
-                    .beginTransaction()
-                    .replace(R.id.frame_container, mDetailCategoryFragment, DetailCategoryFragment::class.java.simpleName)
-                    .addToBackStack(null)
-                    .commit()
+            val mFragmentManager = fragmentManager
+            mFragmentManager?.beginTransaction()?.apply {
+                replace(R.id.frame_container, mDetailCategoryFragment, DetailCategoryFragment::class.java.simpleName)
+                addToBackStack(null)
+                commit()
+            }
```

Comment:
It should be used immediately
`fragmentManager?.beginTransaction().apply { ... }`

Reply:
error yan klo kayak gitu.. lihat di documentation juga modelnya kayak gini.. https://developer.android.com/guide/components/fragments#Example

Reply:
Di dokumentasi https://developer.android.com/guide/components/fragments#Example kayak gini:
```
                fragmentManager?.beginTransaction()?.apply {
                    if (index == 0) {
                        replace(R.id.details, details)
                    } else {
                        replace(R.id.a_item, details)
                    }
                    setTransition(FragmentTransaction.TRANSIT_FRAGMENT_FADE)
                    commit()
                }
```
Gak liat aku **fragmentManager** ditampung di variable baru.

Reply:
iya.. tak kira nggak pake (?) nya yg dibahas.. 

klo ditampung di variable baru biar seragam kayak yg di MainActivity mungkin (aku ngikut yg sebelumnya..hehe), di situ variabelnya dipake buat findFragmentByTag, selain itu juga biar tau bedanya fragmentManager di Activity sama Fragment

Reply:
Bedain fragment manager yang di Activty sama Fragment gunain **findFragmentByTag** kah? maap aku belum baca lagi modulnya

Reply:
bukan yan.. klo di Fragment pake getFragmentManager

Reply:
Itu hostnya Fragment atau Activity ya Mas @arifaizin ?

Reply:
kalo yg ini di Activity-nya yan